app/presenters/document/field_group/section/value_presenter.rb

```diff
@@ -72,20 +75,13 @@ def linkable_for_entity?
             %w(agents concepts).include?(section.entity_name) # while only agent & concept entity pages are implemented
         end
 
-        def entity_url
-          return nil unless entity.present?
-
-          begin
-            entity_uri = URI.parse(entity.fetch('about'))
-            return nil unless entity_uri.host == 'data.europeana.eu'
-          rescue URI::Error
-            return nil
-          end
-
-          type, namespace, id = entity_uri.path.split('/')[1..-1]
-          return nil unless namespace == 'base'
+        def entity_url(format = 'html')
+          return nil unless  europeana_entity?
+          portal_entity_path(entity.fetch('about'), slug: entity_url_slug(entity), format: format)
+        end
 
-          controller.entity_path(type: entities_human_type(type), id: id, slug: entity_url_slug(entity), format: 'html')
+        def europeana_entity?
+          @europeana_entity ||= for_entity? && europeana_entity_url?(entity.fetch('about'))
```

Comment:
This memoisation is only effective when the stored value is `true`. It will need to be rewritten as something like:
```ruby
return @europeana_entity if instance_variable_defined?(:@europeana_entity)
@europeana_entity = for_entity? && europeana_entity_url?(entity.fetch('about'))
```